internal/rbd/replicationcontrollerserver.go

```diff
@@ -301,9 +296,18 @@ func (rs *ReplicationServer) PromoteVolume(ctx context.Context,
 
 	// promote secondary to primary
 	if !mirroringInfo.Primary {
-		err = rbdVol.promoteImage(force)
+		err = rbdVol.promoteImage(req.Force)
 		if err != nil {
 			util.ErrorLog(ctx, err.Error())
+			// In case of the DR the image on the primary site cannot be
+			// demoted as the cluster is down, during failover the image need
+			// to be force promoted. RBD returns `Device or resource busy`
+			// error message if the image cannot be promoted for above reason.
+			// Return FailedPrecondition so that replication operator can send
+			// request to force promote the image.
+			if strings.Contains(err.Error(), "Device or resource busy") {
```

Comment:
@Madhu-1  I understand,  this is validated at PromoteImage scenario here, however  just wondering is there a chance for some other scenario which could cause `Device or resource busy` error to be returned on an image while attempted. May be @dillman can confirm

Reply:
I have seen this error when the promotion is not possible because the image was primary on another cluster and cannot be promoted without the force option.